qiskit/pulse/parser.py

```diff
@@ -303,3 +310,13 @@ def parse_string_expr(source: str, partial_binding: bool = False):
         source = source.replace(match, sub)
 
     return PulseExpression(source, partial_binding)
+
+
+def _match_Num_or_Parameter(node: ast.AST) -> bool:
```

Comment:
Let's stick to snake case
```suggestion
def _match_num_or_parameter(node: ast.AST) -> bool:
```

Reply:
This is the convention for AST functions. For example https://docs.python.org/3/library/ast.html#ast.NodeTransformer